### PR TITLE
GMAC demo program.

### PR DESCRIPTION
**Pull request by paulidale**
Add a GMAC demonstration program.

### CODE REVIEW

File: demos/evp/gmac.c [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    unsigned char outbuf[1024];
+
+    printf("AES GMAC:\n");
+    printf("Authenticated Data:\n");
+    BIO_dump_fp(stdout, gmac_aad, sizeof(gmac_aad));
+
+    ctx = EVP_CIPHER_CTX_new();
+
+    /* Set cipher type and mode */
+    EVP_EncryptInit_ex(ctx, EVP_aes_128_gcm(), NULL, NULL, NULL);
```
Comment (romen) [inline demos/evp/gmac.c]: I understand it's a demo example, but I believe we should lead by example and always check return values.

Comment (paulidale) [reply demos/evp/gmac.c]: I've added error checking and diagnostics.

**Comment by paulidale** [comment]
Many of the demos skip the error checking -- something we should look at sometime.

**Comment by paulidale** [comment]
Merged, thanks.

**Pull request closed without merging**